src/phase0/state_transition/epoch_processing.rs

```diff
@@ -154,8 +156,48 @@ pub fn process_registry_updates<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    // Process activation eligibility and ejections
+    let current_epoch = get_current_epoch(state, context);
+
+    for i in 0..state.validators.len() {
+        let validator = &mut state.validators[i];
+        if is_eligible_for_activation_queue(validator, context) {
+            validator.activation_eligibility_epoch = current_epoch + 1;
+        }
+
+        if is_active_validator(validator, current_epoch)
+            && validator.effective_balance <= context.ejection_balance
+        {
+            initiate_validator_exit(state, i, context);
+        }
+    }
+
+    // Queue validators eligible for activation and not yet dequeued for activation
+    let mut activation_queue = state
+        .validators
+        .iter()
+        .enumerate()
+        .filter(|&(_, validator)| is_eligible_for_activation(state, validator))
```

Comment:
`filter` then `map` here

Reply:
this isn't the end of the world but it does extend the iterator chain so there is some performance hit...

the idea with filter map:

```rust
.filter_map(|&(index, validator)| {
    if is_eligible(validator) {
        Some(index)
    } else {
        None
    }
})
```

but not a huge deal

Reply:
Ah, right, change makes sense then -- done